mysql_fdw.c

```diff
@@ -942,11 +950,15 @@ mysqlGetForeignPlan(PlannerInfo *root,RelOptInfo *baserel, Oid foreigntableid, F
 	 * field of the finished plan node; we can't keep them in private state
 	 * because then they wouldn't be subject to later planner processing.
 	 */
-	return make_foreignscan(tlist,
-							local_exprs,
-							scan_relid,
-							params_list,
-							fdw_private);
+	return make_foreignscan(tlist
```

Comment:
I am getting the error below when trying to compile the mysql_fdw with PG
9.5.

Ibrar is going to commit a fix for this later today.

make USE_PGXS=1
gcc -Wall -Wmissing-prototypes -Wpointer-arith
-Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute
-Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard
-O2 -fpic -I/home/edb/mysql-5.6.19-linux-glibc2.5-x86_64/include -D
_MYSQL_LIBNAME=\"libmysqlclient.so\" -I. -I./
-I/home/edb/Projects/PG_servers/postgres/binary/include/server
-I/home/edb/Projects/PG_servers/postgres/binary/include/internal
-D_GNU_SOURCE   -c -o connection.o connection.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith
-Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute
-Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard
-O2 -fpic -I/home/edb/mysql-5.6.19-linux-glibc2.5-x86_64/include -D
_MYSQL_LIBNAME=\"libmysqlclient.so\" -I. -I./
-I/home/edb/Projects/PG_servers/postgres/binary/include/server
-I/home/edb/Projects/PG_servers/postgres/binary/include/internal
-D_GNU_SOURCE   -c -o option.o option.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith
-Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute
-Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard
-O2 -fpic -I/home/edb/mysql-5.6.19-linux-glibc2.5-x86_64/include -D
_MYSQL_LIBNAME=\"libmysqlclient.so\" -I. -I./
-I/home/edb/Projects/PG_servers/postgres/binary/include/server
-I/home/edb/Projects/PG_servers/postgres/binary/include/internal
-D_GNU_SOURCE   -c -o deparse.o deparse.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith
-Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute
-Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard
-O2 -fpic -I/home/edb/mysql-5.6.19-linux-glibc2.5-x86_64/include -D
_MYSQL_LIBNAME=\"libmysqlclient.so\" -I. -I./
-I/home/edb/Projects/PG_servers/postgres/binary/include/server
-I/home/edb/Projects/PG_servers/postgres/binary/include/internal
-D_GNU_SOURCE   -c -o mysql_query.o mysql_query.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith
-Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute
-Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard
-O2 -fpic -I/home/edb/mysql-5.6.19-linux-glibc2.5-x86_64/include -D
_MYSQL_LIBNAME=\"libmysqlclient.so\" -I. -I./
-I/home/edb/Projects/PG_servers/postgres/binary/include/server
-I/home/edb/Projects/PG_servers/postgres/binary/include/internal
-D_GNU_SOURCE   -c -o mysql_fdw.o mysql_fdw.c
mysql_fdw.c: In function ‘mysqlGetForeignPlan’:
mysql_fdw.c:949:8: error: too few arguments to function ‘make_foreignscan’
In file included from mysql_fdw.c:41:0:
/home/edb/Projects/PG_servers/postgres/binary/include/server/optimizer/planmain.h:46:21:
note: declared here
mysql_fdw.c: In function ‘mysqlPlanForeignModify’:
mysql_fdw.c:1073:35: error: ‘RangeTblEntry’ has no member named
‘modifiedCols’
mysql_fdw.c: In function ‘mysqlGetForeignPlan’:
mysql_fdw.c:950:1: warning: control reaches end of non-void function
[-Wreturn-type]
make: *** [mysql_fdw.o] Error 1


On Sat, Jun 27, 2015 at 4:18 AM, jjthiessen <notifications@github.com>
wrote:

> In mysql_fdw.c
> <https://github.com/EnterpriseDB/mysql_fdw/pull/62#discussion_r33404303>:
>
> > @@ -942,11 +950,15 @@ mysqlGetForeignPlan(PlannerInfo *root,RelOptInfo *baserel, Oid foreigntableid, F
> >  	 * field of the finished plan node; we can't keep them in private state
> >  	 * because then they wouldn't be subject to later planner processing.
> >  	 */
> > -	return make_foreignscan(tlist,
> > -							local_exprs,
> > -							scan_relid,
> > -							params_list,
> > -							fdw_private);
> > +	return make_foreignscan(tlist
>
> I switched the formatting around to allow for capturing the change (within
> the #if) as locally as possible while maintaining consistent formatting
> across the items in the argument list of make_foreignscan.
>
> I'm also fine with only using a start of line comma on the line within the
> #if (maintaining previous formatting/trailing commas elsewhere); or
> encapsulating the whole return in a #if/#else macro block (possibly
> cleaner looking, but more duplicated lines).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/EnterpriseDB/mysql_fdw/pull/62/files#r33404303>.
>



-- 
Ahsan Hadi
Snr Director Product Development
EnterpriseDB Corporation
The Enterprise Postgres Company

Phone: +92-51-8358874
Mobile: +92-333-5162114

Website: www.enterprisedb.com
EnterpriseDB Blog: http://blogs.enterprisedb.com/
Follow us on Twitter: http://www.twitter.com/enterprisedb

This e-mail message (and any attachment) is intended for the use of the
individual or entity to whom it is addressed. This message contains
information from EnterpriseDB Corporation that may be privileged,
confidential, or exempt from disclosure under applicable law. If you are
not the intended recipient or authorized to receive this for the intended
recipient, any use, dissemination, distribution, retention, archiving, or
copying of this communication is strictly prohibited. If you have received
this e-mail in error, please notify the sender immediately by reply e-mail
and delete this message.